### ISSUE TITLE
Multi objective vs. single objective with combined metrics

### ISSUE THREAD

**Issue by brunofacca** [optuna/optuna#2237]
Thanks for this great library.

I'm working with a dataset that has an extremely low signal-to-noise ratio and a relatively small number of observations, so it is extremely susceptible to overfitting. Even though I use KFold CV and there are no information leaks, I recently had inflated validation scores due to overfitting of the *validation* data. Using nested CV would be too expensive and may not completely solve the problem. I've thought of an unusual way of penalizing sets of hyperparameters that are more prone to overfitting: penalize high differences between train and validation score. I use MCC, so a higher validation score is better. My question is, is there a difference between using a single objective study to maximize something like  `val_score / (train_score / val_score)` vs. using a multi-objective study and passing it the validation score (as an objective to maximize) and the difference between the  training and the validation score (as an objective to minimize) separately? Would the multi-objective study handle the separate objectives in a smarter way?

Thank you.

**Comment by nabenabe0928** [comment]
Hallo, I will post some points here.

First, multi-objective optimization tries to find the so-called Pareto front shown below.
![Dominated-non-dominated-and-Pareto-front-solution-set](https://user-images.githubusercontent.com/47781922/105138900-e39dd500-5b38-11eb-8ce9-cd8149dcab47.png)

On the other hand, the single-objective function tries to find the single objective function in a straightforward way.

Basically, the Pareto front includes the optimal values for each objective.
Plus, you can visualize the trade-off between one objective and another.
In this sense, if you are interested in the trade-off between validation error and generalization score (, which you defined), multi-objective optimization can be an option.

The clear advantage of the single-objective is sample-efficiency and the disadvantage of the single-objective is the opaqueness of the result.
If you would like to observe diverse configurations and pick the one you think is the best, I would suggest you to take multi-objective.
However, if your concern still lies in the computational budget, you should take single-objective.

**Comment by brunofacca** [comment]
Thank you very much for the detailed reply. 

I think I'll stick with the single objective for now due to computational budget concerns.
